app/classifier/tasks/survey/index.spec.js

```diff
@@ -1,72 +1,162 @@
   });
 
   describe('with an existing annotation', function () {
-    const annotation = {
-      value: [{
-        choice: 'ar',
-        answers: {
-          ho: 'two',
-          be: [
-            'ea'
-          ]
-        },
-        filters: {}
-      }]
-    };
     const task = workflow.tasks.survey;
     let wrapper;
 
     beforeEach(function () {
-      wrapper = mount(<SurveyTask translation={task} task={task} annotation={annotation} />);
+      wrapper = mount(
+        <SurveyTask
+          annotation={annotation}
+          onChange={onChangeSpy}
+          translation={task}
+          task={task}
+        />
+      );
     });
 
     it('should render a valid annotation as a selected choice', function () {
       const selectedChoice = wrapper.find('[data-choiceid="ar"]');
-      assert.equal(selectedChoice.prop('className'), 'survey-task-chooser-choice-button survey-task-chooser-choice-button-chosen');
+      expect(selectedChoice.prop('className'), 'survey-task-chooser-choice-button survey-task-chooser-choice-button-chosen').to.be.equal;
     });
 
     it('should pass existing answers to the Choice component', function () {
       wrapper.setState({ selectedChoiceID: 'ar' });
       const choice = wrapper.find(Choice);
-      assert.equal(choice.props().annotationValue, annotation.value[0]);
+      expect(choice.props().annotationValue, annotation.value[0]).to.be.equal;
     });
 
     it('should reset saved choices when the annotation resets', function () {
       wrapper.setState({ selectedChoiceID: 'ar' });
       let choice = wrapper.find(Choice);
-      assert.equal(choice.props().annotationValue.answers.ho, 'two');
-      assert.equal(choice.instance().state.answers.ho, 'two');
+      expect(choice.props().annotationValue.answers.ho, 'two').to.be.equal;
+      expect(choice.instance().state.answers.ho, 'two').to.be.equal;
       wrapper.setState({ selectedChoiceID: '' });
       wrapper.setProps({ annotation: { value: [] }});
       wrapper.update();
       wrapper.setState({ selectedChoiceID: 'ar' });
       choice = wrapper.find(Choice);
-      assert.equal(choice.props().annotationValue.answers.ho, undefined);
-      assert.equal(choice.instance().state.answers.ho, undefined);
+      expect(choice.props().annotationValue.answers.ho, undefined).to.be.equal;
+      expect(choice.instance().state.answers.ho, undefined).to.be.equal;
+    });
+  });
+
+  describe('functions', function() {
+    let clock;
+    let wrapper = shallow(
+      <SurveyTask
+        annotation={annotation}
+        onChange={onChangeSpy}
+        translation={task}
+        task={task}
+      />
+    );
+
+    before(function () {
+      const date = new Date();
+      clock = sinon.useFakeTimers(date.getTime());
+    });
+
+    it('should correctly call handleAnnotation', function() {
+      wrapper.instance().handleAnnotation(selection.choice, selection.answers);
+      expect(onChangeSpy.calledOnce).to.be.true;
+      clock.tick();
```

Comment:
Exactly. Without `clock.tick()` the test cannot be accurately executed because the function and test both finish before the items inside `setTimeout` execute.